File: xmrspacebanxxserver/minerBTC/trade.js
import ccxt from "ccxt";
import { exec } from "child_process";

// ======================
// Configuración
// ======================
const SYMBOL = "USDC/WLD";   // Par de trading
const TIMEFRAME = "1m";      // Velas de 1 minuto
const LIMIT = 50;            // Número de velas
const SLEEP = 60 * 1000;     // 60 segundos
const SMA_SHORT = 5;         // Media móvil rápida
const SMA_LONG = 20;         // Media móvil lenta

let running = true;

// ======================
// Conexión al exchange
// ======================
const exchange = new ccxt.binance({
  enableRateLimit: true,
});

// ======================
// Funciones auxiliares
// ======================
async function fetchOHLCV() {
  const ohlcv = await exchange.fetchOHLCV(SYMBOL, TIMEFRAME, undefined, LIMIT);
  return ohlcv.map((candle) => ({
    time: candle[0],
    close: candle[4],
  }));
}

function SMA(data, length) {
  return data.slice(-length).reduce((a, b) => a + b, 0) / length;
}

function beep(freq = 880, duration = 0.2) {
  exec(`play -nq -t alsa synth ${duration} sine ${freq}`);
}

function vibrate(ms = 500) {
  exec(`termux-vibrate -d ${ms}`);
}

async function tradingSignal() {
  const data = await fetchOHLCV();
  const closes = data.map((d) => d.close);

  const smaShortNow = SMA(closes.slice(-SMA_SHORT), SMA_SHORT);
  const smaLongNow = SMA(closes.slice(-SMA_LONG), SMA_LONG);

  const smaShortPrev = SMA(closes.slice(-(SMA_SHORT + 1), -1), SMA_SHORT);
  const smaLongPrev = SMA(closes.slice(-(SMA_LONG + 1), -1), SMA_LONG);

  const lastPrice = closes[closes.length - 1];

  if (smaShortNow > smaLongNow && smaShortPrev <= smaLongPrev) {
    console.log(`🚀 Señal de COMPRA a ${lastPrice}`);
    vibrate(500);
    beep(880, 0.2);
  } else if (smaShortNow < smaLongNow && smaShortPrev >= smaLongPrev) {
    console.log(`⛔ Señal de VENTA a ${lastPrice}`);
    vibrate(1000);
    beep(440, 0.2);
  } else {
    console.log(`🟡 Sin señal. Precio actual: ${lastPrice}`);
  }
}

// ======================
// Loop principal
// ======================
console.log(`📊 Bot de análisis iniciado para ${SYMBOL}`);
console.log("ℹ️ Escribe Ctrl + C para finalizar.");

(async function loop() {
  while (running) {
    try {
      await tradingSignal();
    } catch (err) {
      console.error("❌ Error:", err);
    }
    await new Promise((res) => setTimeout(res, SLEEP));
  }
})();

File: xmrspacebanxxserver/minerBTC/trade.py
import time
import pandas as pd
import ccxt
import os
import threading

# ======================
# Configuración
# ======================
SYMBOL = 'WLD/USDT'        # Par de trading
TIMEFRAME = '1m'           # Velas de 1 minuto
LIMIT = 50                 # Número de velas para análisis
SLEEP = 60                 # Tiempo entre análisis (segundos)
SMA_SHORT = 5              # Media móvil rápida
SMA_LONG = 20              # Media móvil lenta

# Variable global para controlar el loop
running = True

# ======================
# Conexión al exchange
# ======================
exchange = ccxt.binance({'enableRateLimit': True})

def fetch_ohlcv():
    """Descargar velas y devolver DataFrame"""
    ohlcv = exchange.fetch_ohlcv(SYMBOL, timeframe=TIMEFRAME, limit=LIMIT)
    df = pd.DataFrame(ohlcv, columns=['timestamp','open','high','low','close','volume'])
    df['close'] = df['close'].astype(float)
    return df

def trading_signal(df):
    """Analizar señales de compra/venta con medias móviles"""
    df['sma_short'] = df['close'].rolling(SMA_SHORT).mean()
    df['sma_long'] = df['close'].rolling(SMA_LONG).mean()
    
    sma_short_now = df['sma_short'].iloc[-1]
    sma_long_now = df['sma_long'].iloc[-1]
    sma_short_prev = df['sma_short'].iloc[-2]
    sma_long_prev = df['sma_long'].iloc[-2]
    last_price = df['close'].iloc[-1]

    if sma_short_now > sma_long_now and sma_short_prev <= sma_long_prev:
        print(f"🚀 Señal de COMPRA a {last_price}")
        os.system("termux-vibrate -d 500")
        os.system("play -nq -t alsa synth 0.2 sine 880")
    elif sma_short_now < sma_long_now and sma_short_prev >= sma_long_prev:
        print(f"⛔ Señal de VENTA a {last_price}")
        os.system("termux-vibrate -d 1000")
        os.system("play -nq -t alsa synth 0.2 sine 440")
    else:
        print(f"🟡 Sin señal. Precio actual: {last_price}")

def user_input_listener():
    """Hilo para escuchar si el usuario escribe 'exit'"""
    global running
    while running:
        cmd = input()
        if cmd.lower() == "exit":
            print("👋 Finalizando bot por comando del usuario...")
            running = False
            break

# ======================
# Loop principal
# ======================
print(f"📊 Bot de análisis iniciado para {SYMBOL}")
print("ℹ️ Escribe 'exit' y presiona ENTER para finalizar el bot.")

# Iniciar hilo que escucha comandos del usuario
threading.Thread(target=user_input_listener, daemon=True).start()

try:
    while running:
        df = fetch_ohlcv()
        trading_signal(df)
        time.sleep(SLEEP)
except KeyboardInterrupt:
    print("\n👋 Finalizando bot por Ctrl + C...")

